homeassistant/components/template/alarm_control_panel.py

```diff
@@ -0,0 +1,322 @@
+"""Support for Template alarm control panels."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.components.alarm_control_panel import (
+    ENTITY_ID_FORMAT, FORMAT_NUMBER, AlarmControlPanel)
+from homeassistant.const import (
+    CONF_VALUE_TEMPLATE, CONF_ICON_TEMPLATE, CONF_ENTITY_PICTURE_TEMPLATE,
+    CONF_ENTITY_ID, CONF_FRIENDLY_NAME, STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME, STATE_ALARM_DISARMED, STATE_ALARM_TRIGGERED,
+    STATE_ALARM_ARMED_NIGHT, EVENT_HOMEASSISTANT_START, MATCH_ALL, ATTR_CODE
+)
+from homeassistant.helpers.config_validation import PLATFORM_SCHEMA
+from homeassistant.exceptions import TemplateError
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import async_generate_entity_id
+from homeassistant.helpers.event import async_track_state_change
+from homeassistant.helpers.script import Script
+
+_LOGGER = logging.getLogger(__name__)
+_VALID_STATES = [STATE_ALARM_ARMED_AWAY, STATE_ALARM_ARMED_HOME,
+                 STATE_ALARM_DISARMED, STATE_ALARM_TRIGGERED,
+                 STATE_ALARM_ARMED_NIGHT]
+
+CONF_ARM_AWAY_ACTION = 'arm_away'
+CONF_ARM_HOME_ACTION = 'arm_home'
+CONF_ARM_NIGHT_ACTION = 'arm_night'
+CONF_DISARM_ACTION = 'disarm'
+CONF_ALARM_CONTROL_PANELS = 'panels'
+CONF_CODE_ARM_REQUIRED = 'code_arm_required'
+
+ALARM_CONTROL_PANEL_SCHEMA = vol.Schema({
+    vol.Optional(CONF_VALUE_TEMPLATE): cv.template,
+    vol.Optional(CONF_DISARM_ACTION): cv.SCRIPT_SCHEMA,
+    vol.Optional(CONF_ARM_AWAY_ACTION): cv.SCRIPT_SCHEMA,
+    vol.Optional(CONF_ARM_HOME_ACTION): cv.SCRIPT_SCHEMA,
+    vol.Optional(CONF_ARM_NIGHT_ACTION): cv.SCRIPT_SCHEMA,
+    vol.Optional(CONF_ICON_TEMPLATE): cv.template,
+    vol.Optional(CONF_ENTITY_PICTURE_TEMPLATE): cv.template,
+    vol.Optional(CONF_CODE_ARM_REQUIRED, default=True): cv.boolean,
+    vol.Optional(CONF_FRIENDLY_NAME, default="Template Alarm"): cv.string,
+    vol.Optional(CONF_ENTITY_ID): cv.entity_ids
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ALARM_CONTROL_PANELS):
+        cv.schema_with_slug_keys(ALARM_CONTROL_PANEL_SCHEMA),
+})
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Set up the Template Alarm Control Panels."""
+    alarm_control_panels = []
+
+    for device, device_config in config[CONF_ALARM_CONTROL_PANELS].items():
+        friendly_name = device_config.get(CONF_FRIENDLY_NAME, device)
```

Comment:
Please use `dict[key]` for required config keys and keys with default config schema values.